private/buf/cmd/buf/command/alpha/plugin/pluginpush/pluginpush.go

```diff
@@ -160,57 +161,52 @@ func run(
 	if err != nil {
 		return err
 	}
-
-	// TODO: Now that the imageDigest is resolved, create a bufplugin.Plugin,
-	// then map it into a *registryv1alpha1.RemotePlugin
-	// (or *registryv1alpha1.CreateRemotePluginRequest) so that it can be pushed
-	// to the BSR.
-	//
-	//  plugin, err := bufplugin.NewPlugin(
-	//    pluginConfig.PluginVersion(),
-	//    pluginConfig.Options(),
-	//    pluginConfig.Runtime(),
-	//    imageDigest,
-	//  )
-	//  if err != nil {
-	//    return err
-	//  }
-	//  protoPlugin, err := bufplugin.PluginToProtoPlugin(plugin)
-	//  if err != nil {
-	//    return err
-	//  }
-	//  // Use the RemotePluginService (shown below) ...
-	//
-	// ---
-	//
-	// TODO: At this point, it's possible that an OCI registry entry was created
-	// without successfully creating the metadata in the BSR. If the user tries
-	// to push again, the OCI registry entry might already exist. We need to explore
-	// the OCI registry API to see what we can do to prevent such cases. This might
-	// involve some combination of a cleanup process, a two-phase commit flow (i.e.
-	// first create the metadata in the BSR, then push to the OCI registry, etc), or
-	// something else entirely. This might not actually be an issue depending on the
-	// OCI registry API.
-	//
-	//
+	plugin, err := bufplugin.NewPlugin(
+		pluginConfig.PluginVersion,
+		pluginConfig.Options,
+		pluginConfig.Runtime,
+		buildResponse.Digest,
+	)
+	if err != nil {
+		return err
+	}
 	apiProvider, err := bufcli.NewRegistryProvider(ctx, container)
 	if err != nil {
 		return err
 	}
-	service, err := apiProvider.NewPluginService(ctx, pluginConfig.Name.Remote())
+	service, err := apiProvider.NewPluginCurationService(ctx, pluginConfig.Name.Remote())
 	if err != nil {
 		return err
 	}
-	// TODO: Refactor this to be a new endpoint that uses the protoPlugin
-	// object. This is simply left as a placeholder.
-	if _, err := service.CreatePlugin(
+	var nextRevision int32
+	currentRevision, err := service.GetLatestCuratedPlugin(ctx, pluginConfig.Name.Owner(), pluginConfig.Name.Plugin(), pluginConfig.PluginVersion)
+	if err != nil {
+		if connect_go.CodeOf(err) != connect_go.CodeNotFound {
+			return err
+		}
+		nextRevision = 0
+	} else {
+		nextRevision = currentRevision.Revision + 1
+		if nextRevision < 0 {
+			return fmt.Errorf("next plugin revision out of range")
```

Comment:
Probably overthinking this (and we can also validate we don't wrap over on the server side (no negative revisions).

Reply:
As discussed offline, we might even want to require that the revision is part of the `plugin_version` so that the client doesn't need to perform the `GetLatestCuratedPlugin` call.

This is fine for now though, especially because its already written in this state - we can start with this and change it later if need be.

Maybe just add in a `TODO` here that outlines some of this comment so that we remember to revisit it.